tests/Agent/UnitTests/Core.UnitTest/Errors/ErrorServiceTests.cs

```diff
@@ -317,6 +317,79 @@ public void FromException_MarkErrorDataAsExpected_SameErrorClass_In_ExpectedClas
 
         }
 
+        [TestCase("test group")]
+        [TestCase(null)]
```

Comment:
Shouldn't `errorData.ErrorGroup` come back `null` for all three of these test cases?

Reply:
I'm keeping the abstraction very thin at this point and enforcing the data at the attribute level. So The AttributeDefinition has a conversion function to enforce that the invalid data is coerced to null in a single location. I considered putting it in the ErrorService, but it just seemed too easy to bypass the ErrorService and create an ErrorData instance directly. In fact a lot of the existing tests bypassed the ErrorService, which led to inconsistent behavior between tests.